DEF-3361 improved profiler ui

danengelbrecht-king opened the pull request:
https://docs.google.com/document/d/1oPGELK8ZtgX9BS8lhu-OWnpJ_tJdcizX-p1dQhIw-1k/edit?usp=sharing

NOTE: A new library "profilerext_null" has been added and if you remove the "profilerext" lib in your app manifest you should add "profilerext_null" as a replacement. This was not needed to do before, but now it is.

danengelbrecht-king committed:
New Profiler UI

danengelbrecht-king committed:
Use filtered elapsed time for sorting in profiler instead of peak

danengelbrecht-king committed:
Fixed const-issue in hashtable.h

danengelbrecht-king committed:
Restructured how profile_render stores data
Added pause/resume/steponmaxframetime mode to profile render
Added recording, stepping, max-frame step and hiding

danengelbrecht-king committed:
Added HashTable::Put return parameter

Now returns the pointer to the stored value avoiding Put followed by Get to find the stored value

danengelbrecht-king committed:
Cleaned up profiler draw and added portrait/landscape mode
Fixed sorting when stepping through recorded frames
Fixed getting stuck on frame zero when viewing recorded frames

danengelbrecht-king committed:
Profile render tweaks

danengelbrecht-king committed:
Moved platform specific includes from profile.h to profile.cpp

danengelbrecht-king committed:
Removed dead code and reduced flickering

danengelbrecht-king committed:
Removed DM_COUNTER_HASH - use automatic precompute of hash for all DM_COUNTER

danengelbrecht-king committed:
Fix type in dmProfile::GetNowTicks for win32

danengelbrecht-king committed:
Keep table of names to reduce profile frame size

danengelbrecht-king committed:
Cleanups and simplification

danengelbrecht-king committed:
Cleanup of scope/sameplsum/counter creation

danengelbrecht-king committed:
Make sure we record the unfiltered and unsorted frame

danengelbrecht-king committed:
Cleanup

danengelbrecht-king committed:
Don’t purge when showing peak frame

danengelbrecht-king committed:
More cleanup

danengelbrecht-king committed:
Revert "Added HashTable::Put return parameter"

This reverts commit 9add9e1c21ade3afdad170c96f4013d9ad51c6f6.

danengelbrecht-king committed:
Removed dead code

danengelbrecht-king committed:
Fixed broken SampleSum sorting

danengelbrecht-king committed:
Bumped profile ui space to better match engine setup of profiler

danengelbrecht-king committed:
Pre-calculate text render batch key

Don’t let dmRender::DrawText calculate batch key hash for each call
Minor cleanups

danengelbrecht-king committed:
More cleanup

danengelbrecht-king committed:
Some comments and cleanups

danengelbrecht-king committed:
Made sorting a more lightweight operation

Sorting is done on indirection table.
Increased filtering on sample values to smooth out sorting which is now done every frame

danengelbrecht-king committed:
Improved sort order index removal

danengelbrecht-king committed:
Cleanup after merge

danengelbrecht-king committed:
Merge branch 'dev' into DEF-3361-Improved-profiler-UI

danengelbrecht-king committed:
Merge remote-tracking branch 'origin/dev' into DEF-3361-Improved-profiler-UI

danengelbrecht-king committed:
Fixed DynamicScope test

danengelbrecht-king committed:
Cleanups

danengelbrecht-king committed:
Reverted whitespace change

danengelbrecht-king committed:
Moved constants together

danengelbrecht-king committed:
Merge remote-tracking branch 'origin/dev' into DEF-3361-Improved-profiler-UI

danengelbrecht-king committed:
Faster free of scopes, samplesums and counters

Also, sorting of counters added

danengelbrecht-king committed:
State fixes

danengelbrecht-king committed:
Tweaked stats sizes

danengelbrecht-king committed:
Small cleanups and comments

mathiaswking reviewed (approved):
I think the code will be better understood if some of the features/rationale was transferred from the design document to the code.

I'm also curious to see some numbers on added overhead.
It is after all a profiling library. :)

Also, please squash this merge! ;)

danengelbrecht-king committed:
clang format

danengelbrecht-king committed:
Use dmIndexPool and group data section in RenderProfile

danengelbrecht-king committed:
Let hash table allocate memory itself to reduce complexity

danengelbrecht-king committed:
Merged origin/dev

danengelbrecht-king committed:
Make sure we get correct sample times on out of profile span

Async threads may create/destroy scopes outside of dmProfile::Begin()/dmProfile::End()
Make sure we measure the elapsed time using only thread local measurements

danengelbrecht-king committed:
Don’t measure time we wait on blocking message dispatch

danengelbrecht-king committed:
Issue log warning if a single profile scope takes more than three seconds and profiler is enabled

danengelbrecht-king committed:
Broke out render_profile messages in engine DispatchProfileRenderMessages

danengelbrecht-king committed:
Allow for longer sample names when in landscape mode

danengelbrecht-king committed:
Sample name truncation on script function

danengelbrecht-king committed:
First draft doc of new profiler functions

danengelbrecht-king committed:
GetSampleSumHash -> GetCombinedHash

danengelbrecht-king committed:
Removed obsolete ResetFreeIndexes()

danengelbrecht-king committed:
Removed obsolete HASHTABLE_BUFFER_SIZE

danengelbrecht-king committed:
Added some inline overview comments

danengelbrecht-king committed:
Make sure we use all of SAMPLE_FRAMES_NAME_LENGTH

danengelbrecht-king committed:
Rename SampleSum to SampleAggregate

danengelbrecht-king committed:
Corrections to docs

danengelbrecht-king committed:
Move profile_render to engine/render module

danengelbrecht-king committed:
Use profilerext_null lib where appropriate

danengelbrecht-king committed:
Make sure symbol ProfilerExt is present in profiler_null

danengelbrecht-king committed:
Remove if !defined(DM_RELEASE) for dmProfiler calls

danengelbrecht-king committed:
Removed obsolete comment

danengelbrecht-king committed:
Merge remote-tracking branch 'origin/dev' into DEF-3361-Improved-profiler-UI

danengelbrecht-king committed:
Merge remote-tracking branch 'origin/dev' into DEF-3361-Improved-profiler-UI

danengelbrecht-king committed:
Added test to make sure null version of profilerext has correct symbols

danengelbrecht-king committed:
Exclude profiler tests if —skip-build-tests is set

danengelbrecht-king committed:
Remove disable_ui and use “enabled” flag in enable_ui. Instead

danengelbrecht-king committed:
profiler.enable_ui example in docs

danengelbrecht-king committed:
Removed bad dependencies for profilerext_null test

danengelbrecht-king committed:
Reverted unneeded whitespace changes

danengelbrecht-king committed:
Reverted one more whitespace change

danengelbrecht-king committed:
More whitespace and comment fixes

mathiaswking reviewed (dismissed):
Looks (mostly) good.
Needs one change and some clarifications.

danengelbrecht-king committed:
Clean up render profile if disable_ui is not called before exit

danengelbrecht-king committed:
Review cleanups and remove profiler.flush_recorded_frames

danengelbrecht-king committed:
Cleanup silly dead code/code duplication

danengelbrecht-king committed:
Cleanup

danengelbrecht-king committed:
Show recorded frame fixes

danengelbrecht-king committed:
Make sure scopes does not overwrite counters in landscape mode
